### PR TITLE
Do not clone allocas in PeepholeOpt

### PR DESCRIPTION
**Pull request by kuhar**
This is meant to fix the issue #215.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by JacobHeAMD** [comment]
ok to test

**Comment by JacobHeAMD** [comment]
test this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4fc7bbb1c48c59ef3d96a656b56eb42b969fcc83
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 88d997710b4e405f3a8e3fd60a38afee9e3e77e2</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 2cb5558b94c5dc839e093cb439057a1802426c8e</li>
<li>LLVM: amd-vulkan-dev 1fc1a7d4248b4749c3df21eb48f7ae97b6cddf74</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 927/927 (100.0%)</li>
<li>Failed: 0/927 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 68331/68331 (100.0%)</li>
<li>Failed: 0/68331 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 68363/68363 (100.0%)</li>
<li>Failed: 0/68363 (0.0%)</li>
</ul></details>


</ul></details>

File: patch/llpcPatchPeepholeOpt.cpp [hunk 1]
```diff
@@ -655,7 +655,9 @@ void PatchPeepholeOpt::visitPHINode(
         }
     }
 
-    if (pPrevIncomingInst != nullptr)
+
```
Comment (amdrexu) [inline patch/llpcPatchPeepholeOpt.cpp]: Remove this line

File: patch/llpcPatchPeepholeOpt.cpp [hunk 2]
```diff
@@ -655,7 +655,9 @@ void PatchPeepholeOpt::visitPHINode(
...
+    // Do not clone allocas -- we don't want to potentially introduce them in the middle of the function.
+    if (pPrevIncomingInst != nullptr && !isa<AllocaInst>(pPrevIncomingInst))
```
Comment (amdrexu) [inline patch/llpcPatchPeepholeOpt.cpp]: if (pPrevIncomingInst != nullptr && !isa<AllocaInst>(pPrevIncomingInst))
->
if ((pPrevIncomingInst != nullptr) && (isa<AllocaInst>(pPrevIncomingInst) == false))

(1) Each condition with relational statement ought to have a "( )"
(2) We don't use negative expression

**Comment by amdrexu** [comment]
Please use git commit --amend -C head to update the commit. Don't make a new commit to fix review issues. Always keep one commit in a pull request. Thank you.

**Comment by kuhar** [comment]
Formatting fixed.

**Comment by amdrexu** [comment]
Once review issues are addressed, please click "resolve conversation" button in the pull-request webpage. Thank you.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9ef82466d2658c40bf241a1089fb95266201f9b3
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 88d997710b4e405f3a8e3fd60a38afee9e3e77e2</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 2cb5558b94c5dc839e093cb439057a1802426c8e</li>
<li>LLVM: amd-vulkan-dev 1fc1a7d4248b4749c3df21eb48f7ae97b6cddf74</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 927/927 (100.0%)</li>
<li>Failed: 0/927 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 68331/68331 (100.0%)</li>
<li>Failed: 0/68331 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 68363/68363 (100.0%)</li>
<li>Failed: 0/68363 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**